File: src/Application Specific/Classes/Entity Collections/Invoiced/Invoiced.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace Invoiceds

        Public Class Invoiced

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "Invoiced Properties"


            Private _InvoicedID As Integer = 0
            Public ReadOnly Property InvoicedID() As Integer
                Get
                    Return _InvoicedID
                End Get
            End Property
            Public WriteOnly Property SetInvoicedID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_InvoicedID", Value)
                End Set
            End Property


            Private _InvoiceNumber As String = String.Empty
            Public ReadOnly Property InvoiceNumber() As String
                Get
                    Return _InvoiceNumber
                End Get
            End Property
            Public WriteOnly Property SetInvoiceNumber() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_InvoiceNumber", Value)
                End Set
            End Property


            Private _AdhocInvoiceType As String = String.Empty
            Public ReadOnly Property AdhocInvoiceType() As String
                Get
                    Return _AdhocInvoiceType
                End Get
            End Property
            Public WriteOnly Property SetAdhocInvoiceType() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_AdhocInvoiceType", Value)
                End Set
            End Property

            ' AdhocInvoiceType

            Private _RaisedDate As DateTime = Nothing
            Public Property RaisedDate() As DateTime
                Get
                    Return _RaisedDate
                End Get
                Set(ByVal Value As DateTime)
                    _RaisedDate = Value
                End Set
            End Property


            Private _RaisedByUserID As Integer = 0
            Public ReadOnly Property RaisedByUserID() As Integer
                Get
                    Return _RaisedByUserID
                End Get
            End Property
            Public WriteOnly Property SetRaisedByUserID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_RaisedByUserID", Value)
                End Set
            End Property

            Private _VATRateID As Integer = 0
            Public ReadOnly Property VATRateID() As Integer
                Get
                    Return _VATRateID
                End Get
            End Property
            Public WriteOnly Property SetVATRateID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_VATRateID", Value)
                End Set
            End Property



            Private _PaymentTermID As Integer = 0
            Public ReadOnly Property PaymentTermID() As Integer
                Get
                    Return _PaymentTermID
                End Get
            End Property
            Public WriteOnly Property SetPaymentTermID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PaymentTermID", Value)
                End Set
            End Property


            Private _PaidByID As Integer = 0
            Public ReadOnly Property PaidByID() As Integer
                Get
                    Return _PaidByID
                End Get
            End Property
            Public WriteOnly Property SetPaidByID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PaidByID", Value)
                End Set
            End Property


#End Region

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Invoiced/Invoiced.cs
using System;
using System.Collections;

namespace FSM.Entity
{
    namespace Invoiceds
    {
        public class Invoiced
        {
            private DataTypeValidator _dataTypeValidator;

            public Invoiced()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            private int _InvoicedID = 0;

            public int InvoicedID
            {
                get
                {
                    return _InvoicedID;
                }
            }

            public object SetInvoicedID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_InvoicedID", value);
                }
            }

            private string _InvoiceNumber = string.Empty;

            public string InvoiceNumber
            {
                get
                {
                    return _InvoiceNumber;
                }
            }

            public object SetInvoiceNumber
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_InvoiceNumber", value);
                }
            }

            private string _AdhocInvoiceType = string.Empty;

            public string AdhocInvoiceType
            {
                get
                {
                    return _AdhocInvoiceType;
                }
            }

            public object SetAdhocInvoiceType
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_AdhocInvoiceType", value);
                }
            }

            // AdhocInvoiceType

            private DateTime _RaisedDate = default;

            public DateTime RaisedDate
            {
                get
                {
                    return _RaisedDate;
                }

                set
                {
                    _RaisedDate = value;
                }
            }

            private int _RaisedByUserID = 0;

            public int RaisedByUserID
            {
                get
                {
                    return _RaisedByUserID;
                }
            }

            public object SetRaisedByUserID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_RaisedByUserID", value);
                }
            }

            private int _VATRateID = 0;

            public int VATRateID
            {
                get
                {
                    return _VATRateID;
                }
            }

            public object SetVATRateID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_VATRateID", value);
                }
            }

            private int _PaymentTermID = 0;

            public int PaymentTermID
            {
                get
                {
                    return _PaymentTermID;
                }
            }

            public object SetPaymentTermID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PaymentTermID", value);
                }
            }

            private int _PaidByID = 0;

            public int PaidByID
            {
                get
                {
                    return _PaidByID;
                }
            }

            public object SetPaidByID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PaidByID", value);
                }
            }


            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}